pages/leaderboard/[metricId].js

```diff
@@ -0,0 +1,106 @@
+import { GraphQLClient } from 'graphql-request'
+import Head from 'next/head'
+import { useRouter } from 'next/router'
+import React from 'react'
+
+import { DateRangePicker, Layout, Loader } from '../../components'
+import { MetricQuery } from '../../graphql'
+import { useDataFetching } from '../../utils'
+
+const client = new GraphQLClient(
+  process.env.NEXT_PUBLIC_GRAPHQL_ENDPOINT_US_EAST_2
+)
+
+const TODAY_DATE = new Date()
+const THIRTY_DAYS_AGO = new Date(TODAY_DATE - 30 * 24 * 60 * 60 * 1000)
+
+export default function LeaderboardPage() {
+  const [uniqueName, setUniqueName] = React.useState()
+  const [description, setDescription] = React.useState()
+  const [startDate, setStartDate] = React.useState(THIRTY_DAYS_AGO)
+  const [stopDate, setStopDate] = React.useState(TODAY_DATE)
+
+  const router = useRouter()
+  const { metricId } = router.query
+
+  const { isLoading, accessToken, metric } = useDataFetching(
+    client,
+    metricId,
+    MetricQuery
+  )
+
+  React.useEffect(() => {
+    setUniqueName(metric?.uniqueName)
+    setDescription(metric?.description)
+  }, [metric])
+
+  return (
+    <>
+      <Head>
+        <title>Propel Sample Leaderboard</title>
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <Layout
+        title={{
+          url: 'https://www.propeldata.com/docs/metrics/leaderboard',
+          text: 'Leaderboard'
+        }}
+      >
+        {isLoading ? (
+          <Loader />
+        ) : (
+          <>
+            <h4 style={{ marginTop: '2rem' }}>{uniqueName}</h4>
+            <p>{description}</p>
+            <DateRangePicker
+              startDate={startDate}
+              stopDate={stopDate}
+              setStartDate={setStartDate}
+              setStopDate={setStopDate}
+            />
+            {uniqueName && (
+              <Leaderboard
+                variant="bar"
+                styles={{
+                  bar: {
+                    thickness: 15,
+                    borderWidth: '0',
+                    borderRadius: 2,
+                    borderColor: '#94A3B8',
+                    backgroundColor: '#6941C6',
+                    hoverBackgroundColor: 'rgba(158, 119, 237, 0.60)'
+                  },
+                  canvas: {
+                    height: 400
+                  }
+                }}
+                query={{
+                  metric: uniqueName,
+                  accessToken,
+                  timeRange: {
+                    start: format(startDate, 'yyyy-MM-dd'),
+                    stop: format(stopDate, 'yyyy-MM-dd')
+                  },
+                  rowLimit: 8,
+                  dimensions: [
+                    {
+                      columnName: 'sauce_name'
+                    },
+                    {
+                      columnName: 'taco_name'
+                    },
+                    {
+                      columnName: 'restaurant_name'
+                    }
+                  ]
```

Comment:
Rather than hard-code, could we make these dynamic? The reason is, these dimension names are first of all specific to the TacoSoft demo data, but second of all, must be pre-configured on the chosen Metric.

The way we handle this in our Console is by allowing the user to choose which dimensions to include, and the dimensions are populated dynamically based on the Metric's dimensions:

<img width="411" alt="image" src="https://github.com/propeldata/propel-nextjs-starter-app/assets/359154/4cde374c-2975-4700-9826-b97841f50cdc">

If there is at least one dimension available, a leaderboard can be visualized. If there are no dimensions, the leaderboard cannot be visualized.

Probably this type of component should eventually be extracted into the UI Kit itself. 🤔

Reply:
I agree, since we are using MUi here as well maybe we can do something similar to what we do in Console.

Reply:
I agree, since we are using MUi here maybe we can do something similar to what we do in Console.

Reply:
@sashathor I added you to our Console project so that you can see the code @felipecadavid mentioned. @felipecadavid, I think you'll need to help share that component.

Reply:
@sashathor the component is [here](https://github.com/propeldata/console/blob/main/src/features/playground/widgets/LeaderboardBuilder/LeaderboardBuilder.tsx#L228-L282)